pkgs/firehose/lib/firehose.dart

```diff
@@ -23,6 +34,174 @@ class Firehose {
+// Copyright (c) ${DateTime.now().year}, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.''';
+
+    var filePaths = await _getFilesWithoutLicenses(github);
+
+    var markdownResult = '''
+Some `.dart` files were found to not have license headers. Please add the following header to all listed files:
+```
+$license
+```
+
+| Files |
+| :--- |
+${filePaths.map((e) => '|$e|').join('\n')}
+
+Either manually or by running the following in your repository directory
+
+```
+dart pub global activate --source git https://github.com/mosuem/file_licenser
+dart pub global run file_licenser .
+```
+
+'''; //TODO: replace by pub.dev version
+
+    return HealthCheckResult(
+      _licenseBotTag,
+      filePaths.isNotEmpty ? Severity.error : Severity.success,
+      markdownResult,
+    );
+  }
+
+  Future<HealthCheckResult> changelogCheck(Github github) async {
+    var filePaths = await _packagesWithoutChangelog(github);
+
+    final markdownResult = '''
+Changes to these files need to be accounted for in their respective changelogs:
+
+| Package | Files |
+| :--- | :--- |
+${filePaths.entries.map((e) => '| package:${e.key.name} | ${e.value.map((e) => path.relative(e, from: Directory.current.path)).join('<br />')} |').join('\n')}
+''';
+
+    return HealthCheckResult(
+      _changelogBotTag,
+      filePaths.isNotEmpty ? Severity.error : Severity.success,
+      markdownResult,
+    );
+  }
+
+  Future<Map<Package, List<String>>> _packagesWithoutChangelog(
+      Github github) async {
+    final repo = Repository();
+    final packages = repo.locatePackages();
+
+    final files = await github.listFilesForPR();
+    print('Collecting packages without changed changelogs:');
+    final packagesWithoutChangedChangelog = packages.where((package) {
+      var changelogPath = package.changelog.file.path;
+      var changelog =
+          path.relative(changelogPath, from: Directory.current.path);
+      return !files.contains(changelog);
+    }).toList();
+    print('Done, found ${packagesWithoutChangedChangelog.length} packages.');
+
+    print('Collecting files without license headers in those packages:');
+    var packagesWithChanges = <Package, List<String>>{};
+    for (final file in files) {
+      for (final package in packagesWithoutChangedChangelog) {
+        if (fileNeedsEntryInChangelog(package, file)) {
+          print(file);
+          packagesWithChanges.update(
+            package,
+            (changedFiles) => [...changedFiles, file],
+            ifAbsent: () => [file],
+          );
+        }
+      }
+    }
+    print('''
+Done, found ${packagesWithChanges.length} packages with a need for a changelog.''');
+    return packagesWithChanges;
+  }
+
+  bool fileNeedsEntryInChangelog(Package package, String file) {
+    //TODO: Add conditions, such as file is not a markdown etc. Find out what is
```

Comment:
Likely something like everything in lib/ and bin/, and README.md and pubspec.yaml. Or we could just exclude known items - test/ and .github?